packages/muelu/src/Smoothers/BlockedSmoothers/MueLu_BlockedGaussSeidelSmoother_def.hpp

```diff
@@ -302,6 +302,15 @@ namespace MueLu {
     LocalOrdinal nSweeps = pL.get<LocalOrdinal>("Sweeps");
     Scalar omega = pL.get<Scalar>("Damping factor");
 
+
+    // Clear solution from previos V cycle if still stored
+    Scalar normX = rcpX->getVector(0)->norm2();
+    if(InitialGuessIsZero==true && normX > 0){
+      rcpX->putScalar(Teuchos::ScalarTraits<Scalar>::zero());
+      // Is there a better way to do this?
+    }
```

Comment:
@pohm01 What about this?
```c++
    if (InitialGuessIsZero)
      rcpX->putScalar(Teuchos::ScalarTraits<Scalar>::zero());
```
Instead of computing the norm (which involves some local operations and some communication), why not just set all entries to zero. The norm seems not to be required for anything else.

Alternatively, you could compute the norm only in case of `InitialGuessIsZero == true):
```c++
    if (InitialGuessIsZero) {
      Scalar normX = rcpX->getVector(0)->norm2();
      if (normX > 0)
        rcpX->putScalar(Teuchos::ScalarTraits<Scalar>::zero());
    }
```

These are just ideas and suggestions. While I like the approach without ever computing the norm the most, of course I leave the decision up to you.

Reply:
There actually were already tests for each smoother with and without a zero initial guess. I've added a third that does a zero initial guess without zeroing the solution vector `X` that is passed in.